src/core/Merger.cpp

```diff
@@ -632,7 +633,9 @@ Merger::ChangeList Merger::mergeMetadata(const MergeContext& context)
 
         // Check missing keys from source. Remove those from target
         for (const auto& key : targetCustomDataKeys) {
-            if (!sourceMetadata->customData()->contains(key)) {
+            // Do not remove any browser keys
+            if (!sourceMetadata->customData()->contains(key) && !key.startsWith(BrowserService::ASSOCIATE_KEY_PREFIX)
+                && !key.startsWith(CustomData::Created)) {
```

Comment:
I don't like how specific this is to the browser plugin. I would prefer a new function in the Merger class, perhaps named `isProtectedCustomData(key)`, that does these checks and is separate from the mechanics of the merge.

What is the long-term solution to this as more and more plugins use the custom data?

Reply:
That would be much nicer, yes.

For the long-term solution I would suggest creating a new class for a certain type of custom data that will have Created/Modified timestamps (and a possible prefix), so merging identical key/values could be compared, and the implementation would be slightly more generic.